File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/core/RowCodecBasic.scala
package dev.typr.foundations.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*


import java.time.Instant

@SuppressWarnings(Array("unused"))
object RowCodecBasic:
  //start
  case class Person(id: Int, name: String, createdAt: Instant)

  val personParser: RowCodec[Person] = RowCodec.builder[Person]()
    .field(PgTypes.int4)(_.id)
    .field(PgTypes.text)(_.name)
    .field(PgTypes.timestamptz)(_.createdAt)
    .build(Person.apply)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/RowCodecBasic.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.PgTypes;
import dev.typr.foundations.RowCodec;

import java.time.Instant;

@SuppressWarnings("unused")
public class RowCodecBasic {
    //start
    record Person(Integer id, String name, Instant createdAt) {}

    RowCodec<Person> personParser =
        RowCodec.<Person>builder()
            .field(PgTypes.int4, Person::id)
            .field(PgTypes.text, Person::name)
            .field(PgTypes.timestamptz, Person::createdAt)
            .build(Person::new);
    //stop
}
